shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -867,7 +867,14 @@ static bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
 
   FML_DCHECK([[NSThread currentThread] isMainThread]);
```

Comment:
thx for the chat just now. just to check if i understand it correctly (let me know if i miss anything): 

Dart layer sends separate messages (info about `views_to_dispose` and `composition_order`) to the engine. The `composition_order` contains what should be rendered on the current frame, and `views_to_dispose` contains views that needs to be disposed. We need separate messages because a platform view could be just hidden and not disposed, so it won't show up in `composition_order`. 

The crash happens when `views_to_dispose` message is delivered earlier than the new `composition_order` (due to the async nature of the message channel). So when rendering the current frame, a platform view is already disposed, but it still exists in `composition_order`. The fix here is simply to delay the `views_to_dispose` if it's still part of `composition_order`, and try again next frame. Alternative fix is to remove the view from `composition_order`, but we decided that `composition_order` should be the source of truth of the current frame.

Reply:
Yes.